modules/foreman_subnet.py

```diff
@@ -0,0 +1,315 @@
+    from: "192.168.0.2"
+    to: "192.168.0.42"
+    boot_mode: "Static"
+    dhcp_proxy: "smart-proxy1.foo.example.com"
+    tftp_proxy: "smart-proxy1.foo.example.com"
+    dns_proxy: "smart-proxy2.foo.example.com"
+    vlanid: 452
+    mtu: 9000
+    domains:
+    - "foo.example.com"
+    - "bar.example.com"
+    organizations:
+    - "Example Org"
+    locations:
+    - "Toulouse"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    verify_ssl: False
+    state: present
+'''
+
+RETURN = ''' # '''
+
+try:
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        ping_server,
+        find_subnet,
+        find_domains,
+        find_locations,
+        find_organizations,
+        find_smart_proxy,
+        find_smart_proxies,
+        naildown_entity_state,
+        sanitize_entity_dict,
+    )
+
+    from nailgun.entities import (
+        Subnet,
+    )
+
+    has_import_error = False
+except ImportError as e:
+    has_import_error = True
+    import_error_msg = str(e)
+
+from netaddr import IPNetwork
+from ansible.module_utils.basic import AnsibleModule
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'network_type': 'network_type',
+    'dns_primary': 'dns_primary',
+    'dns_secondary': 'dns_secondary',
+    'domains': 'domain',
+    'gateway': 'gateway',
+    'network': 'network',
+    'cidr': 'cidr',
+    'mask': 'mask',
+    'from_ip': 'from',
+    'to_ip': 'to',
+    'boot_mode': 'boot_mode',
+    'ipam': 'ipam',
+    'dhcp_proxy': 'dhcp',
+    'tftp_proxy': 'tftp',
+    'discovery_proxy': 'discovery',
+    'dns_proxy': 'dns',
+    'remote_execution_proxies': 'remote_execution_proxy',
+    'vlanid': 'vlanid',
+    'mtu': 'mtu',
+    'organizations': 'organization',
+    'locations': 'location',
+}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(type='bool', default=True),
+            name=dict(required=True),
+            network_type=dict(choices=['IPv4', 'IPv6'], default='IPv4'),
+            dns_primary=dict(),
+            dns_secondary=dict(),
+            domains=dict(type='list'),
+            gateway=dict(),
+            network=dict(required=True),
+            cidr=dict(type='int'),
+            mask=dict(),
+            from_ip=dict(),
+            to_ip=dict(),
+            boot_mode=dict(choices=['DHCP', 'Static'], default='DHCP'),
+            ipam=dict(choices=['DHCP', 'Internal DB'], default='DHCP'),
+            dhcp_proxy=dict(),
+            tftp_proxy=dict(),
+            discovery_proxy=dict(),
+            dns_proxy=dict(),
+            remote_execution_proxies=dict(type='list'),
+            vlanid=dict(type='int'),
+            mtu=dict(type='int'),
```

Comment:
Since MTU isn't supported, I'd suggest we either remove it or add support to Nailgun

Reply:
Yes, I Know, I would like to add this to Nailgun. This on my TODO for today to see how to contribute to nailgun for this :)